engine/flink/management/src/main/scala/pl/touk/nussknacker/engine/management/FlinkConfig.scala

```diff
@@ -1,32 +1,36 @@
 package pl.touk.nussknacker.engine.management
 
-import scala.concurrent.duration.FiniteDuration
-import scala.concurrent.duration.DurationInt
+import pl.touk.nussknacker.engine.api.deployment.cache.ScenarioStateCachingConfig
+
+import scala.concurrent.duration.{DurationInt, FiniteDuration}
 
 /**
   * FlinkConfig deployment configuration.
   *
   * @param restUrl REST API endpoint of the Flink cluster.
+  * @param scenarioStateCaching Flink's scenario state caching configuration, to see more check {@link ScenarioStateCachingConfig}.
   * @param jobManagerTimeout Timeout for communication with FLink cluster. Consider extending if e.g. you have long savepoint times etc.
   * @param shouldVerifyBeforeDeploy By default, before redeployment of scenario with state from savepoint, verification of savepoint compatibility is performed. There are some cases when it can be too time consuming or not possible. Use this flag to disable it.
   * @param shouldCheckAvailableSlots When true {@link FlinkDeploymentManager} checks if there are free slots to run new job. This check should be disabled on Flink Kubernetes Native deployments, where Taskmanager is started on demand.
   */
-case class FlinkConfig(
+final case class FlinkConfig(
     restUrl: String,
+    scenarioStateCaching: ScenarioStateCachingConfig = ScenarioStateCachingConfig(),
```

Comment:
`FlinkConfig` is a flink specific config, don't put here general DM's concept of `ScenarioStateCachingConfig`. We should pass this information next to `FlinkConfig`